text/0000-mandatory-unique-identifier-for-telemetry-sources.md

```diff
@@ -0,0 +1,66 @@
+* An UI can use it for visualization, especially as fallback if no other attribute is provided for that.
+* The collector (and other processors) can use that identifier while processing traces, metrics, logs.
+* An end-user could use that identifier for error handling and debugging, e.g. when a telemetry source is mis-configured, it's easier to identify it among others.  
+
+## Internal details
+
+As stated above, there are multiple approaches to obtain that common unique identifier. Depending on the approach, there are different ways to accomplish it:
+
+1. Introduce `telemetry.sdk.instance_id` (or similar) and make it mandatory. Make `service.name` only mandatory for backend services. Other telemetry sources can make different attributes mandatory, like `app.name`. Optionally, remove `service.instance_id` from `service.*`
+
+2. Introduce a broad definition of the term _Service_ in the glossary. Unique identification could be achieved by (1) or making `service.name`, `service.namespace`, `service.instance_id` mandatory for all telemetry sources.
+
+3. Narrow down the definition for the term _Service_ to backend services. Make `service.name` only mandatory for backend services. Other telemetry sources can make different attributes mandatory, like `app.name` and provide a definition for their term, like `App` in the glossary. Unique identification could be achieved by (1) or having `(service|app).instance_id` and `(service|app).namespace` made mandatory as well.
+
+4. Introduce `telemetry.source.name`, `telemetry.source.namespace` and `telemetry.source.instance_id`. Make some or all of them mandatory for all telemetry sources. Different telemetry sources can add additional attributes in namespaces like `service.*` and `app.*`.
+
+## Trade-offs and mitigations
+
+All potential approaches provide different trade-offs:
+
+1. This will not introduce any breaking changes.
+
+2. This will not introduce any breaking changes, but end-users might get confused by calling their telemetry a service while they think of it as an app or different (see future possibilities)
+
+3. This may introduce a breaking change with `service.name` being not mandatory anymore in that broad sense. This would need further investigation. Also, this approach might lead to further additional sets of attributes which will be used by different telemetry sources for unique identification (devices, cronjobs, bots, ...)
+
+4. This will introduce a breaking change because `service.name` will be replaced with `telemetry.source.name`. This could be mitigated by a fallback mechanism, e.g. if `telemetry.source.name` is not provided check `service.name`.
+
+This list is not exhaustive, There are potentially more trade-offs per approach.
+
+## Open questions
+
+* What approach provides the most benefit and the least breaking changes to the current specification?
```

Comment:
Can you propose one approach within this OTEP and list the other approaches as Alternatives considered?

It'll be hard for folks to "approve" this without an approach chosen.


This is a great rundown of options, tradeoffs issues.   I think if you pick the option you find best, you'll see people comment pros/cons and find consensus in comments anyway.  If you don't take a position, you're unlikely to see that feedback.

Reply:
Good point, let me take the one that most people hate, so they will bring their arguments.

Seriously: I'll find some time to rewrite the proposal in such a way! thanks!